optuna/integration/lightgbm_tuner/optimize.py

```diff
@@ -598,7 +598,7 @@ def tune_min_data_in_leaf(self):
         # type: () -> None
 
         param_name = "min_child_samples"
-        param_values = [5, 10, 25, 50, 100]
+        param_values = [5, 10, 25, 50, 100]  # type: List[Union[str, float, int, bool, None]]
```

Comment:
The type of candidate points of `optuna.samplers.GridSampler` is `List[Union[str, float, int, bool, None]]`, and `mypy` says `List[int]` is not compatible with it. I think `int` is included by `Union[str, float, int, bool, None]`, and I'm not sure why the error occurs. Please let me know if you have any ideas.

Reply:
I submitted the PR https://github.com/optuna/optuna/pull/1102 to resolve this issue.

Reply:
The change was reverted in https://github.com/optuna/optuna/pull/1076/commits/56a4ea4b00888f0dd228142fb7bc05f55b9ab45c because #1102 resolved the issue.